src/test/com/fulcrologic/rad/database_adapters/sql_test.clj

```diff
@@ -0,0 +1,70 @@
+(ns com.fulcrologic.rad.database-adapters.sql-test
+  (:require
+    [com.fulcrologic.rad.attributes            :as rad.attr :refer [defattr]]
+    [com.fulcrologic.rad.database-adapters.sql :as rad.sql]
+    [clojure.test :refer [deftest testing is]]))
+
+(defattr account-id :account/id :uuid
+  {::rad.attr/identity? true
+   ::rad.sql/schema :production
+   ::rad.sql/tables #{"accounts"}})
+
+(defattr account-name :account/name :string
+  {::rad.sql/schema :production
+   ::rad.sql/tables #{"accounts"}})
+
+;; Derived data
+(defattr account-locked? :account/locked? :boolean
+  {})
+
+(defattr user-id :user/id :uuid
+  {::rad.sql/schema :production
+   ::rad.sql/tables #{"users"}})
+
+(defattr user-name :user/name :string
+  {::rad.sql/schema :production
+   ::rad.sql/tables #{"users"}})
+
+(deftest delta->txs-test
+  (with-redefs [rad.attr/key->attribute {:account/id      account-id
+                                         :account/name    account-name
+                                         :account/locked? account-locked?
+                                         :user/id         user-id
+                                         :user/name       user-name}]
+
+    (let [account-id #uuid "ffffffff-ffff-ffff-ffff-000000000001"
+          user-id    #uuid "ffffffff-ffff-ffff-ffff-000000000002"
+
+          account-delta
+          {[:account/id account-id] #:account{:name {:before "Joe's pancakes"
+                                                     :after  "Jeff's waffles"}}}
+          account-delta-nosql
+          {[:account/id account-id] #:account{:locked? {:before false
+                                                        :after  true}}}
+          multi-idents-delta
+          {[:account/id account-id] #:account{:name {:before "Joe's pancakes"
+                                                     :after "Jeff's waffles"}}
+           [:user/id user-id]       #:user{:name {:before "Alice"
+                                                  :after "Bob"}}}
+
+          accounts-tx {::rad.sql/table  "accounts"
+                       ::rad.sql/schema :production
+                       :tx/action       :sql/update
+                       :tx/attrs        {:account/name "Jeff's waffles"}
+                       :tx/where        {:account/id account-id}}
+
+          users-tx     {::rad.sql/table  "users"
+                        ::rad.sql/schema :production
+                        :tx/action       :sql/update
+                        :tx/attrs        {:user/name "Bob"}
+                        :tx/where        {:user/id user-id}}]
+
+      (testing "It returns a list of updates"
+        (is (= [accounts-tx] (rad.sql/delta->txs account-delta))))
+
+      (testing "It won't try to persist non-sql attributes"
```

Comment:
Try to use positive statements in test descriptions.  It also won't launch a rocket, paint your house, or walk your dog. There are an infinite number of things you could try to prove it "doesn't" do, and positive-worded language ensures you're testing something the logic *actually does*. In this case, I'd word it "Skips attributes that are not targeted at the SQL data store".  It may sound subtle and nitpicky, but it prevents tests that fail to actually test anything useful.

Reply:
No actually that made a lot of sense. Indeed it won't launch a rocket either, TIL